Code/Source/Entity/Terrain/TerrainData.cpp

```diff
@@ -65,19 +65,21 @@ namespace RGL
             }
         }
 
+        UpdateUvs();
+
         m_indices.clear();
-        m_indices.reserve((heightfieldGridColumns - 1) * (heighfieldGridRows - 1) * TrianglesPerSector);
-        for (size_t sectorIndexX = 0LU; sectorIndexX < heightfieldGridColumns - 1; ++sectorIndexX)
+        m_indices.reserve((m_gridColumns - 1) * (m_gridRows - 1) * TrianglesPerSector);
+        for (size_t sectorIndexX = 0LU; sectorIndexX < m_gridColumns - 1; ++sectorIndexX)
         {
-            for (size_t sectorIndexY = 0LU; sectorIndexY < heighfieldGridRows - 1; ++sectorIndexY)
+            for (size_t sectorIndexY = 0LU; sectorIndexY < m_gridRows - 1; ++sectorIndexY)
             {
-                const auto lowerLeft = aznumeric_cast<int32_t>(sectorIndexY + sectorIndexX * heighfieldGridRows);
-                const auto lowerRight = aznumeric_cast<int32_t>(lowerLeft + heighfieldGridRows);
+                const auto lowerLeft = aznumeric_cast<int32_t>(sectorIndexY + sectorIndexX * m_gridRows);
+                const auto lowerRight = aznumeric_cast<int32_t>(lowerLeft + m_gridRows);
                 const auto upperLeft = aznumeric_cast<int32_t>(lowerLeft + 1);
                 const auto upperRight = aznumeric_cast<int32_t>(lowerRight + 1);
 
-                m_indices.emplace_back(rgl_vec3i{ upperLeft, upperRight, lowerLeft });
-                m_indices.emplace_back(rgl_vec3i{ lowerLeft, upperRight, lowerRight });
+                m_indices.emplace_back(rgl_vec3i{ upperLeft, upperRight, lowerRight });
+                m_indices.emplace_back(rgl_vec3i{ upperLeft, lowerLeft, lowerRight });
```

Comment:
The reasoning here was to align the triangles with the diagonal used by the Terrain Gem. This also doesn't impact calculations in the RGL gem since it doesn't differentiate between clockwise and counter-clockwise triangles (it uses the absolute value of the dot product between the normal and the ray direction).

Reply:
It's a simple change and for the sake of the convention we can change it. However, if you look further down in the file of your screenshot you can see that the absolute value of the dot product is used (so the order in fact does not matter).
![image](https://github.com/user-attachments/assets/54424fb5-3127-4756-8371-3c017da9fbaf)